server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java

```diff
@@ -195,7 +318,6 @@ private void moveUnmergedToNew(TsFileResource seqFile) throws IOException {
         ChunkCache.getInstance().clear();
         ChunkMetadataCache.getInstance().clear();
         TimeSeriesMetadataCache.getInstance().clear();
-        FileReaderManager.getInstance().closeFileAndRemoveReader(seqFile.getTsFilePath());
```

Comment:
it is necessary as the old tsfile's cache is invalid

Reply:
okay, I will fix it

Reply:
add this line to your changed code.